### ISSUE TITLE
[Bug]: USB Bochum (Müllmax) does not work?

### ISSUE THREAD

**Issue by casi1801** [mampfes/hacs_waste_collection_schedule#2951]
### I Have A Problem With:

Sensor configuration, The integration in general

### What's Your Problem

got the following yaml when doing the python script:
# waste_collection_schedule source configuration
waste_collection_schedule:
  sources:
    - name: muellmax_de
      args:
        service: Usb
        mm_frm_str_sel: XYZstraße
        mm_frm_hnr_sel: 44791;Innenstadt;11;
If I try to configure my address in the integration mask I always got the following message:
"Die Quelle hat eine ungültige Antwort zurückgegeben: "Got invalid response from the server, please recheck your arguments". Bitte überprüfe die Argumente und versuche es erneut."
What can I do?

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [ ] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [ ] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by ReneNulschDE** [comment]
Could you please share the real streetname? I tested it with:
```
mm_frm_str_sel: Burggrafenstraße
mm_frm_hnr_sel: 44791;Innenstadt;3;

```
and this gives results.
![image](https://github.com/user-attachments/assets/d13ac7bc-e71a-4fd1-be41-0ce827902b59)

Please check the homeassistant.log for any additional information.

**Comment by casi1801** [comment]
It is Freiligrathstraße 55

**Comment by casi1801** [comment]
![IMG_4412](https://github.com/user-attachments/assets/363e96f6-4873-434a-bb58-249240ed3777)

You just put it in the configuration.yaml, or into the Integration mask??

**Comment by ReneNulschDE** [comment]
Put the values out of the wizard into the fields.

![image](https://github.com/user-attachments/assets/33b0aab4-a94e-4ab0-b9a6-96b8bc079c92)

Result:
![image](https://github.com/user-attachments/assets/118b96af-9fa1-4de6-b5a4-b5cf649ef995)

**Comment by casi1801** [comment]
Ok, I get the calendar, when I put the yaml in my configuration.yaml, but I don’t get a waste Collection schedule Integration where I can configure the calendar, when I try to integrale it! See the mask… 
![IMG_4418](https://github.com/user-attachments/assets/52725a79-24cf-442e-98d0-200d6a015b25)

**Comment by ReneNulschDE** [comment]
Is there maybe a space at the the end of the string? Which version of the component and of HA do you use?

Set the logger to debug. Edit your configuration.yaml and add (or merge in case you have logger config already):
```
logger:
  default: warning
  logs:
    custom_components.waste_collection_schedule: debug
    urllib3.connectionpool: debug
```
Restart HA and try it again. Share the relevant homeassistant.log entries.

**Comment by casi1801** [comment]
I put the logger in my configuration.yaml
restart has
tried to integrate Usb
no spaces at the end of the lines!
Version 2.4.0. (same problem with the version 2.3.0)
there are the following entries in the homeassistant.log...
2024-10-31 21:08:52.322 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-10-31 21:08:52.322 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration waste_collection_schedule which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

**Comment by ReneNulschDE** [comment]
There have to be more lines. Please share the complete log or send it to wcs.dev@nulsch.de in case of privacy questions.

**Comment by ReneNulschDE** [comment]
@casi1801: Could you close this issue?

**Comment by casi1801** [comment]
Ok, it works after deleting an entity which has been remained by a former installation…

**Comment by ddorau** [comment]
I ran into the same Problem for the city Düsseldorf and I found out what's causing this. Müllmax is rate limiting at about 10 request for 24h. I think while we're trying to set the sensor up, we exceeded the limit multiple times, then waited a day or two, deleted it, recreated it and since the rate limit was lifted it worked.

That leads to another problem: restarting HA multiple times a day (working on my dashboards at the moment) I run into the rate limit again and the calendar get's purged what equals in an empty schedule == no pickup dates anymore. Is there a way to disable the re-download while restarting and is it possible to extend the redownload to let's say once a month?
